src/host/getset.cpp

```diff
@@ -1028,36 +1036,89 @@ void ApiRoutines::GetLargestConsoleWindowSizeImpl(const SCREEN_INFORMATION& cont
                 return S_OK;
             }
 
-            const auto clipViewport = clip ? Viewport::FromInclusive(*clip) : bufferSize;
+            const auto clipViewport = clip ? Viewport::FromInclusive(*clip).Clamp(bufferSize) : bufferSize;
             const auto sourceViewport = Viewport::FromInclusive(source);
-            Viewport readViewport;
-            Viewport writtenViewport;
+            const auto fillViewport = sourceViewport.Clamp(clipViewport);
 
-            const auto w = std::max(0, sourceViewport.Width());
-            const auto h = std::max(0, sourceViewport.Height());
-            const auto a = static_cast<size_t>(w * h);
-            if (a == 0)
+            writer.BackupCursor();

```

Comment:
DECCRA and DECFRA are documented as not changing the active cursor position. We may be able to skip backing up the cursor in the DECxRA cases

Reply:
Oh, backing up the cursor also backs up the active attributes. We use those in DECFRA. Got it.